src/server/chromium/crPage.ts

```diff
@@ -749,10 +766,16 @@ class FrameSession {
   }
 
   async _onFileChooserOpened(event: Protocol.Page.fileChooserOpenedPayload) {
-    const frame = this._page._frameManager.frame(event.frameId)!;
-    const utilityContext = await frame._utilityContext();
-    const handle = await this._adoptBackendNodeId(event.backendNodeId, utilityContext);
-    this._page._onFileChooserOpened(handle);
+    const frame = this._page._frameManager.frame(event.frameId);
+    if (!frame)
+      return;
+    try {
+      const utilityContext = await frame._utilityContext();
+      const handle = await this._adoptBackendNodeId(event.backendNodeId, utilityContext);
+      this._page._onFileChooserOpened(handle);
+    } catch (e) {
+      // During async processing, frame/context may go away. We should not throw.
+    }
   }
 
   _onDownloadWillBegin(payload: Protocol.Page.downloadWillBeginPayload) {
```

Comment:
before reporting download to users, let's make sure that `this._frameId` is present inside `this._frameManager` (similarly to dialog event)

Reply:
This is a browser-side event, so we should be good. Also, detaching the frame does not cancel the download, so we should report it anyway.